### PR TITLE
Remove doctoc comments from static help text

### PR DESCRIPTION
**Pull request by irinarenteria**
Addresses issue #2302.

This is a rough first go. Any thoughts on splitting the text on a matched pattern? Since these are private functions, I haven't added any tests, but if they're required let me know. I did do a check on the `faq` file to make sure the fix works and it seems to be fine.

### CODE REVIEW

File: src/leiningen/help.clj [hunk 1]
```diff
@@ -66,9 +66,18 @@
     (some #(if (= (symbol subtask-name) (:name (meta %))) %)
           (:subtasks (meta task)))))
 
+(defn- clean-static-help
+  "Returns a string containing help content. Removes doctoc comments if they
+  are present."
+  [help-text]
+   (let [doctoc-text #"<!-- END doctoc generated TOC please keep comment here to allow auto update -->"]
+     (if (boolean (re-find doctoc-text help-text))
```
Comment (danielcompton) [inline src/leiningen/help.clj]: We could use `clojure.string/includes?` here possibly.

**Comment by danielcompton** [comment]
This seems like a reasonable approach to take, rather than trying to fully parse the markdown. It would strip too much text if we ever moved the table of contents further down the page, but I'm not sure if that's worth worrying about.

**Comment by winks** [comment]
> It would strip too much text if we ever moved the table of contents further down the page,

Now that could be easily fixed by reversing the call und using `first`  before `<!-- START doctoc...`, right?

**Comment by danielcompton** [comment]
@winks, Sorry that wasn't very clear, what I'm imagining is if the doc page was changed to something like:

```md
# Tutorial

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
  <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
  **Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
...
<!-- END doctoc generated TOC please keep comment here to allow auto update -->

Leiningen is for automating Clojure projects without setting your hair on fire.
...
```

To do that, the TOC section would need to be selected and removed from the text, and then the surrounding two parts spliced together. I don't think reversing the call would be enough. I don't know if it's worth the effort and more complicated code to handle that, I'll let a core maintainer weigh in on that.

**Comment by winks** [comment]
@danielcompton oops. that was exactly what I tried to say. a logically reversed call in addition ;)

**Comment by technomancy** [comment]
> Since these are private functions, I haven't added any tests, but if they're required let me know. I did do a check on the faq file to make sure the fix works and it seems to be fine.

No need to add a test for this; manual testing is totally OK. This is not code that changes frequently.

This is great. The only change I would like to see is using something other than `re-find` because we're looking for an exact match rather than a regular expression pattern. `clojure.string/includes?` would be perfect.

Thanks!

**Comment by irinarenteria** [comment]
@danielcompton @winks: I agree, it would be better to ensure only the exact DocToc portion is removed regardless of content structure. It might not be worth implementing at this point though, since it seems like the help content doesn’t change frequently. 

@technomancy  That makes sense. I updated that now. Funnily enough, I had to use `re-pattern` because `split` only takes a regex. Probably an indication that this is likely not the most idiomatic Clojure approach.

File: src/leiningen/help.clj (hunk 1 above)
Comment (irinarenteria) [reply src/leiningen/help.clj]: And that eliminates the need for `boolean` too.

**Comment by technomancy** [comment]
Great; thanks! Merging now.

**Comment by irinarenteria** [comment]
Thank you, @technomancy!

**Comment by ehashman** [comment]
:tada:

**Pull request merged**